libraries/AP_Stats/AP_Stats.cpp

```diff
@@ -0,0 +1,108 @@
+#include "AP_Stats.h"
+
+#include <AP_Math/AP_Math.h>
+
+const extern AP_HAL::HAL& hal;
+
+// table of user settable parameters
+const AP_Param::GroupInfo AP_Stats::var_info[] = {
+
+    // @Param: _BOOTCNT
+    // @DisplayName: Boot Count
+    // @Description: Number of times board has been booted
+    // @User: Standard
+    AP_GROUPINFO("_BOOTCNT",    0, AP_Stats, params.bootcount, 0),
+
+    // @Param: _FLTTIME
+    // @DisplayName: Total FlightTime
+    // @Description: Total FlightTime (seconds)
+    // @User: Standard
+    AP_GROUPINFO("_FLTTIME",    1, AP_Stats, params.flttime, 0),
+
+    // @Param: _RUNTIME
+    // @DisplayName: Total RunTime
+    // @Description: Total time autopilot has run
+    // @User: Standard
+    AP_GROUPINFO("_RUNTIME",    2, AP_Stats, params.runtime, 0),
+
+    // @Param: _RESET
+    // @DisplayName: Reset time
+    // @Description: Seconds since epoch since reset (set to 0 to reset)
+    // @User: Standard
+    AP_GROUPINFO("_RESET",    3, AP_Stats, params.reset, 1),
+
+    AP_GROUPEND
+};
+
+void AP_Stats::copy_variables_from_parameters()
+{
+    flttime = params.flttime;
+    runtime = params.runtime;
+    reset = params.reset;
+}
+
+void AP_Stats::init()
+{
+    params.bootcount.set_and_save(params.bootcount+1);
+
+    // initialise our variables from parameters:
+    copy_variables_from_parameters();
+}
+
+
+void AP_Stats::flush()
+{
+    params.flttime.set_and_save(flttime);
+    params.runtime.set_and_save(runtime);
+}
+
+void AP_Stats::update_flighttime()
+{
+    if (_flying_ms) {
+        const uint32_t now = AP_HAL::millis();
+        const uint16_t delta = (now - _flying_ms)/1000;
```

Comment:
Hmmm.  To be a problem, update_flighttime would need to have not been called for 65,000-ish seconds - and it is called every second.  Or am I missing something?